test/fixtures/es-module-loaders/mock-loader.mjs

```diff
@@ -0,0 +1,136 @@
+import {receiveMessageOnPort} from 'worker_threads';
+const mockedModuleExports = new Map();
+let currentMockVersion = 0;
+
+/**
+ * FIXME: this is a hack to workaround loaders being
+ * single threaded for now
+ */
+function doDrainPort() {
+  let msg;
+  while (msg = receiveMessageOnPort(preloadPort)) {
+    onPreloadPortMessage(msg.message);
+  }
+}
+function onPreloadPortMessage({
+  mockVersion, resolved, exports
+}) {
+  currentMockVersion = mockVersion;
+  mockedModuleExports.set(resolved, exports);
+}
+let preloadPort;
+export function globalPreload({port}) {
+  preloadPort = port;
+  port.on('message', onPreloadPortMessage);
+  port.unref();
+  return `(${()=>{
```

Comment:
This string wrapper leads to some awkward internals and interferes with syntax highlighting. Could this instead be a regular function declaration and return fn.toString()?

Reply:
My syntax highlighting works with this XD. I avoid `fn.toString` usually since I don't like calling `.toString` after years of JS and it biting me on occasion, but for an example/test this is fine.

Reply:
updated to use `String()` so readers can see it is a string without reading to the bottom.